src/components/modules/RecentAndTrending/index.tsx

```diff
@@ -0,0 +1,158 @@
+    sectionTitle: string;
+    order?: GridOrder;
+    tag: string;
+}
+
+export default class RecentAndTrending extends PureComponent<Props> {
+    scaffoldProps = (isDesktop: boolean) =>
+    isDesktop
+      ? {
+          containerPadding: 45,
+          overlap: 30
+        }
+      : {
+          containerPadding: CONTAINER_PADDING.MOBILE,
+          overlap: 15
+        };
+
+    MainPostImageProps = (isDesktop: boolean) =>
+    isDesktop
+      ? {
+          imageWidth: 1200,
+          imageHeight: 627
+        }
+      : {
+          imageWidth: 375,
+          imageHeight: 250
+        };
+
+
+    render() {
+        const {
+            sectionColor,
+            postsOne: [mainPost, ...secondaryPosts],
+            postsTwo, 
+            tag
+        } = this.props;
+
+        return (
+            <Responsive>
+            {({ minWidth }) => {
+                const isDesktop = minWidth(BREAKPOINTS.LG);
+
+                const localFontStyles = isDesktop
+                    ? { fontSize: 24, lineHeight: 29 }
+                    : { fontSize: 20, lineHeight: 24 };
+                return (
+                    <Container type='content' style={{flexDirection: isDesktop? 'row' : 'column', justifyContent: 'space-between'}}>
+                      <View style={{width: isDesktop ? '68%' : '100%'}}>
+                        <Text style={[styles.headerText, localFontStyles, {marginLeft: isDesktop ? 0 : '15px'}]}>Recently in {capitalize(tag)}</Text>
+            
+                        <Row style={{}}>
+                            <OverlapScaffold {...this.scaffoldProps(isDesktop)} isDesktop={isDesktop}>
+                                <OverlapScaffold.Main>
+                                    <Post
+                                    layoutVariant='overlay'
+                                    isDesktop={isDesktop}
+                                    center
+                                    {...this.MainPostImageProps(isDesktop)}
+                                    {...mainPost}
+                                    />
+                                </OverlapScaffold.Main>
+                
+                                <OverlapScaffold.Overlap>
+                                    <ModuleBox offsetDirection={OFFSET_DIRECTION.RIGHT} patternColor={sectionColor}>
+                                    {isDesktop ? (
+                                        <PostGrid order={1} posts={secondaryPosts} />
+                                    ) : (
+                                        <PostGrid order={2} itemsPerRow={1} posts={secondaryPosts} />
+                                    )}
+                                    </ModuleBox>
+                                </OverlapScaffold.Overlap>
+                            </OverlapScaffold>
+                        </Row>
+                      </View>
+
+                      <View style={[isDesktop ? styles.trendingDesktopContainer : styles.trendingMobileContainer]}>
```

Comment:
...and this View could be the Trending component. Looking at the isDesktop check, I'd use the renderMobile() & renderDesktop pattern for it.

Reply:
Moved to its own React component